craft_parts/callbacks.py

```diff
@@ -54,6 +59,21 @@ def register_stage_packages_filter(func: FilterCallback) -> None:
     _STAGE_PACKAGE_FILTERS.append(CallbackHook(func, None))
 
 
+def register_configure_overlay(func: ConfigureOverlayCallback) -> None:
```

Comment:
I'm not settled on the callback name (configure_overlay), so if there are better suggestions I'm all ears